service/agentconf.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// This file has routines which can be used for agent specific functionalites related to service files:
+//	- finding all agents in the machine
+//	- create conf file using the machine details
+// 	- write systemd service file and setting links
+// 	- copy all tools and related to agents and setup the links
+// 	- start all the agents
+// These routines can be used by any tools/cmds trying to implement the above functionality as part of the process, eg. juju-updateseries command.
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/arch"
+	"github.com/juju/utils/fs"
+	"github.com/juju/utils/shell"
+	"github.com/juju/utils/symlink"
+	"github.com/juju/version"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/agent/tools"
+	"github.com/juju/juju/juju/paths"
+	"github.com/juju/juju/service/common"
+	"github.com/juju/juju/service/systemd"
+)
+
+// FindAgents find all the agents available in the machine.
+func FindAgents(dataDir string) (machineAgent string, unitAgents []string, errAgentNames []string, err error) {
```

Comment:
We prefer not to name return values unless it's needed - they can make code more confusing, particularly when used with implicit return values. Generally we only use them when we need to do something with returned error values in a defer (like you do further down). 

It doesn't seem like you need the named return values here, can you remove them? machineAgent, unitAgents, errAgentNames and err should just be local variables.